pkg/compiler/switch_test.go

```diff
@@ -0,0 +1,138 @@
+package compiler_test
+
+import (
+	"math/big"
+	"testing"
+)
+
+var switchTestCases = []testCase{
+	{
+		"simple switch success",
+		`package main
+		func Main() int {
+			a := 5
+			switch a {
+			case 5: return 2
+			}
+			return 1
+		}`,
+		big.NewInt(2),
+	},
+	{
+		"simple switch fail",
+		`package main
+		func Main() int {
+			a := 6
+			switch a {
+			case 5:
+				return 2
+			}
+			return 1
+		}`,
+		big.NewInt(1),
+	},
+	{
+		"multiple cases success",
+		`package main
+		func Main() int {
+			a := 6
+			switch a {
+			case 5: return 2
+			case 6: return 3
+			}
+			return 1
+		}`,
+		big.NewInt(3),
+	},
+	{
+		"multiple cases fail",
+		`package main
+		func Main() int {
+			a := 7
+			switch a {
+			case 5: return 2
+			case 6: return 3
+			}
+			return 1
+		}`,
+		big.NewInt(1),
+	},
+	{
+		"default case",
+		`package main
+		func Main() int {
+			a := 7
+			switch a {
+			case 5: return 2
+			case 6: return 3
+			default: return 4
+			}
+			return 1
+		}`,
+		big.NewInt(4),
+	},
+	{
+		"default case",
```

Comment:
Duplicate test? Maybe some non-default case was meant to be here.

Reply:
Yes, it was `empty case before default`. Fixed.